
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../tb-resistance/">
      
      
        <link rel="next" href="../../additional-info/conda/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.7">
    
    
      
        <title>EQTL - LSHTM Omics</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.8608ea7d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#eqtl" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="LSHTM Omics" class="md-header__button md-logo" aria-label="LSHTM Omics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            LSHTM Omics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              EQTL
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../.." class="md-tabs__link">
          
  
  Home

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../introduction/intro-to-linux/" class="md-tabs__link">
          
  
  Introduction

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../advanced/phylogenetics/" class="md-tabs__link">
          
  
  Advanced

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../microbiome/" class="md-tabs__link">
          
  
  Other omics

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../additional-info/conda/" class="md-tabs__link">
          
  
  Additional info

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../presentations/presentations/" class="md-tabs__link">
          
  
  Presentations

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="LSHTM Omics" class="md-nav__button md-logo" aria-label="LSHTM Omics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    LSHTM Omics
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Home
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../connecting-github/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Getting connected
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Introduction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../introduction/intro-to-linux/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Linux
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../introduction/mapping/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Mapping
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../introduction/variant-detection/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Variant detection
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../introduction/assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../introduction/task/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Task
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Advanced
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Advanced
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../advanced/phylogenetics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../advanced/third-generation-sequencing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Third Generation Sequencing
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../advanced/gwas/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genome Wide Association Studies (GWAS)
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Other omics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Other omics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../microbiome/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Microbiomes Practical
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../transcriptomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Transcriptomics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../methylation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Methylation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ml/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Deep learning (Neural Network)
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tb-resistance/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    TB resistance prediction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    EQTL
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    EQTL
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    <span class="md-ellipsis">
      Introduction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exercise-1-eqtl-linked-with-snps" class="md-nav__link">
    <span class="md-ellipsis">
      Exercise 1: eQTL linked with SNPs
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Exercise 1: eQTL linked with SNPs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#preparing-the-data" class="md-nav__link">
    <span class="md-ellipsis">
      Preparing the data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#running-a-cis-eqtl-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Running a cis-eQTL analysis
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#further-exploring-the-results" class="md-nav__link">
    <span class="md-ellipsis">
      Further exploring the results
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#further-exploration" class="md-nav__link">
    <span class="md-ellipsis">
      Further exploration
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Further exploration">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#trans-eqtl-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      trans-eQTL analysis
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#eqtl-linked-with-methylation" class="md-nav__link">
    <span class="md-ellipsis">
      eQTL linked with methylation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      References
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Additional info
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Additional info
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../additional-info/conda/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Conda Environments: Why and How to Use Them
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../additional-info/software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Software
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../additional-info/SettingUpVirtualLinux/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Virtual Box: Setting Up Linux
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../additional-info/faq/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Frequently asked questions
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Presentations
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Presentations
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../presentations/presentations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Presentations
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="eqtl">EQTL</h1>
<h2 id="introduction">Introduction</h2>
<p>An eQTL or expression Quantitative Trait Loci analysis is a statistical analysis through which the association of genomic DNA sequence variants and gene expression levels can be explored. An eQTL is therefore considered a locus that explains a fraction of the gene expression variance found in a phenotype through direct association tests between genetic markers and gene expression levels (Nika &amp; Dermitzakis, 2013).</p>
<p>In the picture below a typical eQTL analysis is represented. Different SNPs within a 1Mb window on a determined genomic location (represented as blue lines) are tested with the expression levels of a given gene in different samples (harbouring and not harbouring the SNPs).</p>
<p><img alt="" src="../../img/eqtl_1.png" /></p>
<p>In the diagram below we can see ploted the expression levels of the samples with the different SNPs and the frequency of appearance of these SNPs. Therefore, both polymorphisms (AG and AA alleles) will be associated with the underexpression of the gene studied compared to the reference allele (GG). Thereby, these SNPs could be considered as eQTLs.</p>
<p><img alt="" src="../../img/eqtl_2.png" /></p>
<p>The eQTLs can be considered as cis-eQTLs or trans-eQTLs depending on the physical distance from the gene they regulate. The cis-eQTLs are in the adjacent regions of the gene they regulate while trans-eQTLs can be located in a distant genomic region.  </p>
<p><img alt="" src="../img/eqtl_3.png" /></p>
<h2 id="exercise-1-eqtl-linked-with-snps">Exercise 1: eQTL linked with SNPs</h2>
<p>To perform an expression quantitative trait loci (eQTL) study linked with SNPs we are going to need two datasets: one with the SNPs we are going to test and the other one with the normalised expression for each gene in each sample. We are going to try to associate SNPs with differential expression, and, to do so, a statistical analysis will be carried out. In this case, we will perform a linear regression. But there are other statistical tests that can be run to study associations, such as ANOVA or Wilcoxon rank-sum test.</p>
<h3 id="preparing-the-data">Preparing the data</h3>
<p>All the materials necessary for this exercise are located in the eqtl directory. You will find there a table with all the SNPs present in the genomic interval 2420631:2920631 (Rv2159c-Rv2590) of M. tuberculosis, and another table with the normalised expression data of the corresponding genes located in that genomic region. We are going to use the same 6 samples previously used for the differential expression analysis in the transcriptomics session, plus 9 additional lineage 1 and lineage 4 samples we have added to the dataset in order to increase the power of the statistical analysis. In total we will be analysing 15 strains, 7 from lineage 1 and 8 from lineage 4.</p>
<p>We can take a look at the tables in the terminal. Just go to the directory and with the "less" command you can view the tables. From the home directory, type:</p>
<div class="highlight"><pre><span></span><code>cd ~/data/eqtl
less snps_eqtl_table.txt
</code></pre></div>
<p>To exit the view of the tables just type "q".</p>
<div class="admonition terminal">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code>POS     REF     ALT     L1_1    L4_2    L1_3    L1_4    L4_5    L4_6    L1_7   gene
2421619 C       A       0       0       1       0       0       0       0       Rv2159c
2421816 A       G       1       0       1       1       0       0       1       Rv2160A
2423386 C       T       0       0       0       0       0       0       0       Rv2162c
2423785 C       T       1       0       1       1       0       0       1       Rv2162c
2423819 C       G       0       0       0       0       0       0       0       Rv2162c
2423822 A       G       0       0       0       0       0       0       0       Rv2162c
2423993 G       C       0       0       0       0       0       0       0       Rv2162c
2424008 G       T       0       1       0       0       0       1       0       Rv2162c
2424864 A       G       1       0       1       1       0       0       1       -
2424908 G       T       0       1       0       0       0       1       0       -
2424925 A       G       1       1       1       1       1       1       1       -
2425023 G       A       0       0       0       0       0       0       0       -
2425097 A       C       1       0       1       1       0       0       1       Rv2163c
2425471 T       C       1       0       1       1       0       0       1       Rv2163c
2426385 C       G       0       0       0       0       0       0       0       Rv2163c
</code></pre></div>
</div>
<p>In the SNPs table, each row represents one SNP across all the samples (columns). The first column indicates the genomic position of the SNP. The second and third ones refer to the reference and alternative nucleotide. The last one indicates whether the SNP falls within a gene (i.e. Rv2159c) or an intergenic region (-). The number 1 in the sample columns represents the presence of the SNP and the number 0 the absence.</p>
<p>The expression table shows the normalised expression levels of each sample (columns) for each gene (rows).</p>
<div class="highlight"><pre><span></span><code>less expression_eqtl_table.txt
</code></pre></div>
<div class="admonition terminal">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code>gene    L1_1    L4_2    L1_3    L1_4    L4_5    L4_6    L1_7    L1_8    L4_9    L1_10
Rv2158c 5.3592  6.5914  5.3647  5.0135  7.282   7.4848  3.6513  5.2832  7.4075  6.2946
Rv2159c 0.9657  6.0428  0.0975  1.2378  7.2741  7.706   1.0288  3.2491  7.0289  1.6385
Rv2160A -0.6193 5.8577  -0.8708 0.8018  6.5427  6.8666  0.2404  2.482   6.0652  -0.5727
Rv2161c 2.683   9.4069  3.2019  3.0181  8.8226  9.4403  4.2647  4.9166  9.3021  2.7425
Rv2162c 4.7906  9.045   5.1631  5.4924  8.4655  9.7019  5.1945  5.6268  9.089   5.1375
Rv2163c 7.5358  7.5722  8.1996  7.4225  7.06    7.3848  8.5059  8.0789  7.5526  7.481
Rv2164c 5.5294  5.6485  6.6888  5.4251  5.5513  5.748   5.9732  6.5069  5.8695  5.8241
Rv2165c 7.3794  7.7184  8.3536  6.9752  6.9572  7.2733  8.1759  7.9825  7.9506  7.2009
Rv2166c 8.452   8.18    9.0769  7.7743  7.3996  7.6419  9.3086  8.7389  7.7578  7.9759
Rv2169c 6.7328  6.5983  7.1653  7.4208  7.6903  7.6155  5.6544  7.3472  7.1758  6.6482
Rv2170  3.0691  2.558   3.4853  3.4285  3.9117  3.7295  2.2728  3.8404  3.7938  3.8026
Rv2171  6.6739  5.8042  6.4029  5.9551  6.0585  6.0265  5.8921  6.1924  6.7047  6.7438
</code></pre></div>
</div>
<p>You should see similar tables to the ones showed above but with the 15 samples.</p>
<h3 id="running-a-cis-eqtl-analysis">Running a cis-eQTL analysis</h3>
<p>We are going to run the analysis in R. So, after activating this practical's conda environment with conda activate eqtl, let's open R in the terminal, just typing:</p>
<div class="highlight"><pre><span></span><code>R
</code></pre></div>
<p>And then we will load the packages we will need for the analysis. We will use a package called MatrixEQTL for the analysis and a package called ggplot2 for plotting some results.</p>
<div class="highlight"><pre><span></span><code>library(MatrixEQTL)
library(ggplot2)
</code></pre></div>
<p>Firstly, we are going to set up the directories where we have the tables we are going to use. These are going to be the same we just saw above but formatted for using with the R package. The folder where you can find them is called "tables", which is in the eqtl directory. Now you should type:</p>
<div class="highlight"><pre><span></span><code>base.dir = &quot;~/data/eqtl/tables/&quot;
snp_file_name = paste(base.dir, &quot;snps_table.txt&quot;, sep=&quot;&quot;)
expression_file_name = paste(base.dir, &quot;expression_table.txt&quot;, sep=&quot;&quot;)
output_file_name = &quot;~/data/eqtl/results_eqtl.txt&quot;
output_file_name.cis = &quot;~/data/eqtl/results_eqtl_cis.txt&quot;
</code></pre></div>
<p>Now we can load the tables for the analysis. We will use the following script:</p>
<div class="highlight"><pre><span></span><code>expr = read.table(expression_file_name, sep=&quot;\t&quot;, header=T, row.names=1)
snp = read.table(snp_file_name, sep=&quot;\t&quot;, header=T, row.names=1)
</code></pre></div>
<p>And to see the first 6 rows of the tables we just need to type:</p>
<div class="highlight"><pre><span></span><code>head(expr)
head(snp)
</code></pre></div>
<p>You should get something like this:</p>
<div class="admonition terminal">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code>&gt; head(expr)
       L1_1   L4_2    L1_3   L1_4   L4_5   L4_6   L1_7   L1_8   L4_9
Rv2158c  5.3592 6.5914  5.3647 5.0135 7.2820 7.4848 3.6513 5.2832 7.4075
Rv2159c  0.9657 6.0428  0.0975 1.2378 7.2741 7.7060 1.0288 3.2491 7.0289
Rv2160A -0.6193 5.8577 -0.8708 0.8018 6.5427 6.8666 0.2404 2.4820 6.0652
Rv2161c  2.6830 9.4069  3.2019 3.0181 8.8226 9.4403 4.2647 4.9166 9.3021
Rv2162c  4.7906 9.0450  5.1631 5.4924 8.4655 9.7019 5.1945 5.6268 9.0890
Rv2163c  7.5358 7.5722  8.1996 7.4225 7.0600 7.3848 8.5059 8.0789 7.5526
        L1_10  L1_11  L4_12   L4_13  L4_14  L4_15
Rv2158c  6.2946 5.7084 7.7072  8.4239 7.1863 7.7056
Rv2159c  1.6385 1.3116 7.7639  8.8317 7.1415 7.7607
Rv2160A -0.5727 0.2881 6.9682  7.5626 6.3030 6.8682
Rv2161c  2.7425 3.0210 9.4097 10.1360 9.1882 9.5215
Rv2162c  5.1375 4.8047 8.6629  9.3841 8.7615 9.4024
Rv2163c  7.4810 7.7594 7.7636  7.6428 7.6123 8.1138
</code></pre></div>
<div class="highlight"><pre><span></span><code>&gt; head(snp)
        L1_1 L4_2 L1_3 L1_4 L4_5 L4_6 L1_7 L1_8 L4_9 L1_10 L1_11 L4_12 L4_13
2421619    0    0    1    0    0    0    0    0    0     1     0     0     0
2421816    1    0    1    1    0    0    1    1    0     1     1     0     0
2423386    0    0    0    0    0    0    0    0    0     0     1     0     0
2423785    1    0    1    1    0    0    1    1    0     1     1     0     0
2423819    0    0    0    0    0    0    0    0    0     0     0     0     0
2423822    0    0    0    0    0    0    0    0    0     0     0     0     0
        L4_14 L4_15
2421619     0     0
2421816     0     0
2423386     0     0
2423785     0     0
2423819     0     0
2423822     0     0
</code></pre></div>
</div>
<p>We already have the tables ready so now we need to set up the parameters for the analysis.</p>
<p>We are going to run a cis-eQTL analysis, which means we are going to test those SNPs that are located within a distance up to 200 bp from the beginning and end of each gene.</p>
<p>The first step is to establish the threshold p-value. The program runs at the same time a trans and a cis eQTL analysis. As we are only interested now in a cis study, we will establish the threshold for the trans p-value in 0, and for the cis p-value in 1 (this means that the output file will contain all the associations up to a p-value of 1).</p>
<div class="highlight"><pre><span></span><code>pvOutputThreshold_cis = 1
pvOutputThreshold_tra = 0
</code></pre></div>
<p>Now we are going to select the statistical model we want to use. In this case we will use linear regression:</p>
<div class="highlight"><pre><span></span><code>useModel = modelLINEAR
</code></pre></div>
<p>The last thing is to load two more tables: one with the positions of the SNPs and other with the positions of the genes, which we will call snpspos and genepos. These two tables are in the directory 'tables', so to load them in our workspace we just need to type:</p>
<div class="highlight"><pre><span></span><code>snpspos &lt;- read.table(paste(base.dir, &quot;snpspos.txt&quot;, sep=&quot;&quot;), sep=&quot;\t&quot;, header=T)
genepos &lt;- read.table(paste(base.dir, &quot;genepos.txt&quot;, sep=&quot;&quot;), sep=&quot;\t&quot;, header=T)
</code></pre></div>
<p>We can take a look at the tables typing:</p>
<div class="highlight"><pre><span></span><code>head(snpspos)
head(genepos)
</code></pre></div>
<p>The last thing to do is to set up a few more parameters, we are not going to go into too much details for the purpose of this practical. Basically they prepare the matrix used for running the analysis and store our analysis and the results. To set this up, just copy the following script:</p>
<div class="highlight"><pre><span></span><code>errorCovariance = numeric()
snp = SlicedData$new()
snp$fileDelimiter = &quot;\t&quot;
snp$fileOmitCharacters = &quot;NA&quot;
snp$fileSkipRows = 1
snp$fileSkipColumns = 1
snp$fileSliceSize = 2000
snp$LoadFile(snp_file_name)
gene = SlicedData$new()
gene$fileDelimiter = &quot;\t&quot;
gene$fileOmitCharacters = &quot;NA&quot;
gene$fileSkipRows = 1
gene$fileSkipColumns = 1
gene$fileSliceSize = 2000
gene$LoadFile(expression_file_name)
cvrt = SlicedData$new()
</code></pre></div>
<p>And finally, we will run the analysis:</p>
<div class="highlight"><pre><span></span><code>me = Matrix_eQTL_main(
    snps = snp,
    gene = gene,
    cvrt = cvrt,
    output_file_name = output_file_name,
    pvOutputThreshold = pvOutputThreshold_tra,
    useModel = useModel,
    errorCovariance = errorCovariance,
    verbose = TRUE,
    output_file_name.cis = output_file_name.cis,
    pvOutputThreshold.cis = 1,
    snpspos = snpspos,
    genepos = genepos,
    cisDist = 200,
    pvalue.hist = TRUE,
    min.pv.by.genesnp = FALSE,
    noFDRsaveMemory = FALSE)
</code></pre></div>
<div class="admonition terminal">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code>Matching data files and location files
439of444 genes matched
847of847 SNPs matched

Task finished in 0.00199999999949796 seconds
Reordering genes

Task finished in 0.231999999999971 seconds
Processing covariates
Task finished in 0 seconds
Processing gene expression data (imputation, residualization)
Task finished in 0.00100000000020373 seconds
Creating output file(s)
Task finished in 0.00399999999990541 seconds
Performing eQTL analysis
100.00% done, 1,055 cis-eQTLs
Task finished in 0.0150000000003274 seconds
</code></pre></div>
</div>
<p>Now let's take a look at the results table. To do it, type the following:</p>
<div class="highlight"><pre><span></span><code>res &lt;- me$cis$eqtls
head(res)
</code></pre></div>
<p>Here we have the table with the 6 top associations found by the analysis just carried out.</p>
<div class="admonition terminal">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code>     snps    gene statistic       pvalue          FDR      beta
1 2423785 Rv2162c -20.05509 3.668713e-11 1.935246e-08 -3.891237
2 2424864 Rv2162c -20.05509 3.668713e-11 1.935246e-08 -3.891237
3 2421816 Rv2160A -13.98178 3.275455e-09 1.151868e-06 -6.379346
4 2421816 Rv2159c -13.41609 5.424159e-09 1.430622e-06 -6.082414
5 2450045 Rv2188c -10.62618 8.853633e-08 1.556764e-05 -3.081468
6 2451081 Rv2188c -10.62618 8.853633e-08 1.556764e-05 -3.081468
</code></pre></div>
</div>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Which is the gene differentially expressed with the highest statistical association? Which is the p-value? And the corresponding adjusted p-value (or FDR)?</p>
</div>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>With how many SNPs is it associated the variable expression of Rv2162c?</p>
</div>
<p>To summarise, we ran a statistical analysis in order to see whether the SNPs present in our samples are associated with differential gene expression. We therefore got a table where we can see the SNP (its position, in snps column) associated with the differential expression of the gene (gene column). A high p-value association is established when a determined SNP is present in all the samples that have a determined gene under- or over-expressed.</p>
<p>However, not all the associations found will be real. cis-eQTLs are more likely to be located within the promoter region of a gene, which is the upstream region where the transcription starts. For instance, a SNP in this promoter region might change the recognition patterns for transcriptional regulators and therefore alter gene expression.</p>
<h3 id="further-exploring-the-results">Further exploring the results</h3>
<p>We will now check one of the resulting genes in the analysis to corroborate if it is a real eQTL association. We are going to take the first gene, Rv2162c. This gene has been associated with 2 SNPs. Let's take a look at the 2 SNPs. To do it, we will load again the SNPs table and look at the rows corresponding to those 2 SNPs:</p>
<div class="highlight"><pre><span></span><code>snps_table &lt;- read.table(&quot;~/data/eqtl/snps_eqtl_table.txt&quot;, sep=&quot;\t&quot;, header=T)
snps_table[snps_table$POS==2423785 | snps_table$POS==2424864,]
</code></pre></div>
<div class="admonition termainl">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code>      POS REF ALT L1_1 L4_2 L1_3 L1_4 L4_5 L4_6 L1_7 L1_8 L4_9 L1_10 L1_11
4 2423785   C   T    1    0    1    1    0    0    1    1    0     1     1
9 2424864   A   G    1    0    1    1    0    0    1    1    0     1     1
L4_12 L4_13 L4_14 L4_15    gene
4     0     0     0     0 Rv2162c
9     0     0     0     0       -
</code></pre></div>
</div>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Which samples carry the SNPs?</p>
<p>Are all of them from the same lineage? Which one?</p>
<p>Are these SNPs located within the coding region or intergenic regions?</p>
</div>
<p>We can see that all the lineage 1 samples harbour the two SNPs. One of them is located in an intergenic region, whilst the other is within the coding region of Rv2162c. As we mentioned before, it is more likely that a SNP present in an upstream or promoter region will affect transcription.</p>
<p>If we go to the website https://mycobrowser.epfl.ch we will find a genomic database for different species of Mycobacterium. We are going to look for our gene of interest: write Rv2162c in the top right white box and select search.</p>
<p><a href="../../img/eqtl_4.png"></a></p>
<p><a href="../../img/eqtl_5.png"></a></p>
<p><a href="../img/eqtl_6.png"></a></p>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Is the gene located in the forward or the reverse strand?</p>
<p>Which is the functional category it belongs to?</p>
</div>
<p>We can see the Rv2162c gene (or PE_PGRS38) represented in purple in the Mycobrowser.</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Is the SNP (A2424864G) in the downstream or upstream region?</p>
</div>
<p>The SNP (A2424864G) is located in the promoter region of the Rv2162c, therefore it is a good candiate for being an eQTL. As the association showed a high p-value, we can expect that in all the samples carrying the SNP (in this case the lineage 1 samples) the Rv2162c gene would be under- or over-expressed. Looking at the expression table we could see this difference in expression, but a more visual way to do it is through a boxplot.</p>
<p>To plot it we need to load a table with the expression of the Rv2162c:</p>
<div class="highlight"><pre><span></span><code>rv2162 &lt;- read.table(&quot;~/data/eqtl/rv2162c.txt&quot;, sep=&quot;\t&quot;, header=T)
rv2162
</code></pre></div>
<div class="admonition terminal">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code>        expg    gene lineage
L1_1  4.7906 Rv2162c       1
L4_2  9.0450 Rv2162c       4
L1_3  5.1631 Rv2162c       1
L1_4  5.4924 Rv2162c       1
L4_5  8.4655 Rv2162c       4
L4_6  9.7019 Rv2162c       4
L1_7  5.1945 Rv2162c       1
L1_8  5.6268 Rv2162c       1
L4_9  9.0890 Rv2162c       4
L1_10 5.1375 Rv2162c       1
L1_11 4.8047 Rv2162c       1
L4_12 8.6629 Rv2162c       4
L4_13 9.3841 Rv2162c       4
L4_14 8.7615 Rv2162c       4
L4_15 9.4024 Rv2162c       4
</code></pre></div>
</div>
<p>And now let's do the plot copying the following script:</p>
<div class="highlight"><pre><span></span><code>rv2162$lineage &lt;- as.factor(rv2162$lineage)
bp &lt;- ggplot(rv2162, aes(x=gene, y=expg, fill=lineage)) +
geom_boxplot() + geom_point(position=position_jitterdodge())
show(bp)
</code></pre></div>
<p>You should get a boxplot like this, where the y axis represents the normalised expression data and the two boxplots are representing the lineage 1 and lineage 4 samples respectively:</p>
<p><a href="../../img/eqtl_7.png"></a></p>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Is Rv2162c differentially expressed between lineage 1 and 4 according to the boxplot?</p>
<p>In which lineage is it less expressed?</p>
<p>Can you think about any possible explanation for the underexpression in the lineage 1 samples carrying the SNP?</p>
</div>
<h2 id="further-exploration">Further exploration</h2>
<h3 id="trans-eqtl-analysis">trans-eQTL analysis</h3>
<p>We have carried out a cis-eQTL analysis, which means that we have only looked at those SNPs close to the genes differentially expressed. However, as we mentioned before, there are also trans-eQTLs. These trans-eQTLs would be, for instance, SNPs located in transcriptional regulators that might alter their function. Hence, transcription of those genes regulated by the altered transcriptional factors could be affected.</p>
<h3 id="eqtl-linked-with-methylation">eQTL linked with methylation</h3>
<p>Methylation is also a mechanism of gene expression control (Casadesus &amp; Low., 2006) and therefore another approach we can take with eQTL analysis is to associate methylation with gene expression. Given the position of the motifs and methylated bases we could treat them as if they were SNPs and perform the same analysis.</p>
<h3 id="references">References</h3>
<p>Nica AC and Dermitzakis ET. (2013). Expression quantitative trait loci: present and future. Philos Trans R Soc Lond B BIol Sci 368(1620):20120362. doi: 10.1098/rstb.2012.0362</p>
<p>Casadesus J and Low D. (2006). Epigenetic gene regulation in the bacterial world. Micobiol Mol Biol Rev 70:830-56.</p>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.tooltips", "navigation.instant", "navigation.tabs", "toc.integrate", "content.tabs.link"], "search": "../../assets/javascripts/workers/search.f8cc74c7.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.c8b220af.min.js"></script>
      
    
  </body>
</html>