
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.7">
    
    
      
        <title>Pylogenetics - LSHTM Omics</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.8608ea7d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#pylogenetics" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="LSHTM Omics" class="md-header__button md-logo" aria-label="LSHTM Omics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            LSHTM Omics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Pylogenetics
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href=".." class="md-tabs__link">
          
  
  Home

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../introduction/intro-to-linux/" class="md-tabs__link">
          
  
  Introduction

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../advanced/phylogenetics/" class="md-tabs__link">
          
  
  Advanced

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../other-omics/microbiome/" class="md-tabs__link">
          
  
  Other omics

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../additional-info/conda/" class="md-tabs__link">
          
  
  Additional info

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../presentations/presentations/" class="md-tabs__link">
          
  
  Presentations

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="LSHTM Omics" class="md-nav__button md-logo" aria-label="LSHTM Omics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    LSHTM Omics
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Home
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../connecting-github/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Getting connected
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Introduction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../introduction/intro-to-linux/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Linux
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../introduction/mapping/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Mapping
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../introduction/variant-detection/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Variant detection
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../introduction/assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../introduction/task/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Task
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Advanced
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Advanced
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../advanced/phylogenetics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../advanced/third-generation-sequencing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Third Generation Sequencing
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../advanced/gwas/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genome Wide Association Studies (GWAS)
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Other omics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Other omics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../other-omics/microbiome/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Microbiomes Practical
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../other-omics/transcriptomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Transcriptomics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../other-omics/methylation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Methylation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../other-omics/ml/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Deep learning (Neural Network)
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../other-omics/tb-resistance/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    TB resistance prediction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../other-omics/eqtl/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    EQTL
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Additional info
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Additional info
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../additional-info/conda/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Conda Environments: Why and How to Use Them
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../additional-info/software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Software
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../additional-info/SettingUpVirtualLinux/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Virtual Box: Setting Up Linux
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../additional-info/faq/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Frequently asked questions
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Presentations
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Presentations
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../presentations/presentations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Presentations
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="pylogenetics">Pylogenetics</h1>
<p>The study of evolutionary relationships among biological entities</p>
<h2 id="objectives">Objectives</h2>
<p>By the end of this practical you should:</p>
<ul>
<li>Understand the relationship between genetic relatedness and transmission events.</li>
<li>Know how to reconstruct and interpret a phylogenetic tree</li>
<li>Understand how time and spatial diffusion of an epidemic is inferred from a phylogeny</li>
</ul>
<h2 id="introduction">Introduction</h2>
<p>In the lecture, we saw how epidemiological processes leave a measurable imprint on pathogen genomes sampled from infected individuals. These processes can be recovered from genetic data sampled at different times and places, using statistical inference methods that take into account the sequences’ shared ancestry.</p>
<p>In this practical session, we will reconstruct and interpret the Influenza A H1N1 2009 (H1N1/09) epidemic in England, based on a set of viral sequences isolated during the outbreak.</p>
<p>We will identify transmission chains of H1N1/09, reconstruct their migration patterns, estimate the number of independent introductions in England, and infer the time and geographical location of these introduction events. In order to do so, we will apply the following procedure:</p>
<ol>
<li>We will reconstruct the phylogeny of influenza genomes sampled in England and other countries where H1N1 infections were diagnosed.</li>
<li>We will apply a molecular clock model to the data in order to fit the phylogeny to real time scales.</li>
<li>We will infer the migration patterns of H1N1 into and within England using Bayesian Markov chain Monte Carlo (MCMC) phylogeographic inference.</li>
</ol>
<hr>

<h2 id="section-1-phylogenetic-reconstruction">Section 1 - Phylogenetic reconstruction</h2>
<p>We will use the packages <strong>AliView</strong> and <strong>RAxML</strong> to view genetic sequence files and build a maximum-likelihood (ML) phylogenetic tree.</p>
<p>The file <strong>‘H1N1.flu.2009.fas’</strong> is a multi-FASTA file containing 50 full-length influenza A H1N1/09 genomes. These nucleotide sequences correspond to influenza viruses sampled in Canada (n = 5), China (n = 3), England (n = 23), Mexico (n = 4), Peru (n = 1), and the United States of America (USA; n = 14) between April and June 2009. The sequences are 12,735 nucleotides long.</p>
<p>We will use AliView to manually inspect the FASTA file:</p>
<p>Activate the correct environment with <code>conda activate phylogenetics</code>.
Open AliView by running the commmand <code>aliview</code>.
Import the sequence file: <strong>File &gt; Open File</strong>, navigate to <strong>~/data/phylogenetics</strong> and select the file <strong>H1N1.flu.2009.fas</strong>.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>To infer the phylogenetic relationship of a set of sequences, these need to be ‘aligned’. That is, nucleotide positions need to be arranged in columns to ensure that we compare homologous positions to one another. In evolutionary biology, homology means similarity due to descent from a common ancestor. Two genes are homologous if they descend from an ancestral gene. Likewise, two nucleotides in different sequences are homologous if they correspond to the same nucleotide position in the ancestral gene. Note that two objects can be ‘similar’ without being ‘identical’. The sequences shown here have been aligned but if your data are not, it will need to be aligned. This can be done in AliView, as well as in other programs including MUSCLE, ClustalW and MAFFT.</p>
</div>
<p>The sequence alignment is displayed as a matrix, where rows correspond to viral samples and columns to nucleotide positions (see <strong>Figure 1</strong>). Cells are coloured by type of nucleotides (A, C, G or T) and missing information, or gaps, are indicated by a dash (“-“).</p>
<p><img alt="Figure 1" src="../img/phylo_1_1.jpg" /></p>
<p>The unique identifier of the sequences is shown at the left hand side of the nucleotide matrix and contains the strain name (e.g. A/England/247/2009) followed by the date of sampling (e.g. d2009-05-31) in the format dYYYY-MM-DD.</p>
<p>In the matrix, we can spot ‘rare’ nucleotide substitutions present in one or more sequences (scroll along the sequence alignment to spot some). These single nucleotide polymorphisms (SNPs) allow us to identify viruses that are genetically related (they share a common SNP) and infer epidemiological linkage between them. If viruses sampled from different individuals have the same SNPs, we can assume that they form a specific strain infecting these individuals. This is the property we use to reconstruct transmission chains from a phylogenetic tree.</p>
<h2 id="tree-reconstruction">Tree reconstruction</h2>
<p><strong>RAxML</strong> [1] is a freely available software used to reconstruct phylogenetic relationships between individuals using a maximum-likelihood (ML) approach. This approach takes into account a substitution model to assess the probability of particular mutations. You can also perform bootstrapping, a method for validating the tree by repeating the analysis a specified number of times to produce pseudoreplicates. This acts as a support for the tree topology of the final tree that is produced by calculating the number of pseudoreplicates in which a given node of a tree is found.</p>
<p>Open your command-line terminal and navigate to the folder containing the <strong>H1N1.flu.2009.fas</strong> sequence file.
<div class="highlight"><pre><span></span><code>cd ~/data/phylogenetics/`
</code></pre></div>
Run the following command (This may take a few minutes to complete):
<div class="highlight"><pre><span></span><code>raxmlHPC -m GTRGAMMA -s H1N1.flu.2009.fas -n H1N1.flu.2009.ML -p 11334 -k -f a -x 13243 -N 100
</code></pre></div></p>
<ul>
<li><strong>raxmlHPC</strong>: This will run RAxML on the command-line and uses the following parameters:</li>
<li><strong>m GTRGAMMA</strong>: This sets the substitution model to be used. We are using GTR (generalised time reversible) with a gamma distribution.</li>
<li><strong>s H1N1.flu.2009.fas</strong>: This is your input file.</li>
<li><strong>n H1N1.flu.2009.ML</strong>: This is the name that will appended to output files.</li>
<li><strong>p 11334</strong>: This is a random seed number.</li>
<li><strong>k</strong>: Trees are printed with branch lengths.</li>
<li><strong>f a</strong>: This tells RAxML to conduct a rapid Bootstrap analysis and search for the best-scoring ML tree in a single run.
-<strong>x 13243</strong>: This is a random seed number for bootstrapping.
-<strong>N 100</strong>: This specifies the number of bootstrap runs.
Once the software has finished running, the file RAxML_bipartitions.H1N1.flu.2009.ML will contain the ML tree with bootstrap supports. Add the extension .tre to the file:</li>
</ul>
<p><div class="highlight"><pre><span></span><code>mv RAxML_bipartitions.H1N1.flu.2009.ML RAxML_bipartitions.H1N1.flu.2009.ML.tre
</code></pre></div>
And open the tree to view in the program FigTree, for this run the command:
<div class="highlight"><pre><span></span><code>figtree
</code></pre></div>
FigTree is a tree editor with a graphic interface and is freely available at <a href="http://tree.bio.ed.ac.uk/software/figtree">http://tree.bio.ed.ac.uk/software/figtree</a>. It runs on all operating systems.</p>
<p>When prompted at opening, name the bootstrap values “bootstrap support”. You can now view the ML tree with bootstrap support values. Click on the <strong>node labels</strong> section on the left of the viewer and select <strong>bootstrap support</strong> from the <strong>Display</strong> drop down box. This will display the bootstrap support for each of the nodes of the tree. Go to <strong>Trees</strong> &gt; <strong>Increasing Node Order</strong> to display the tree by increasing node order.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Note. Many of the values shown are low (the scale is 0 [no support] to 100 [full support]). Typically values over 70% (in this case a bootstrap value of 70) are taken as showing strong support. Low values here may suggest that you need to change the substitution model or run the analysis with more bootstrap runs. For the purposes of time for this exercise, we will continue with the trees produced here.</p>
</div>
<h2 id="tree-interpretation">Tree Interpretation</h2>
<p>Take a few minutes to familiarize yourself with the tree.</p>
<div class="admonition reminder">
<p class="admonition-title">Reminder</p>
<p>The components of a phylogenetic tree are shown in Figure 2. The sequences, or taxa (plural for taxon), are positioned at the end of the external branches. Related sequences are linked by a node (their most recent common ancestor). Two or more sequences descending from a node form a ‘clade’ (or cluster). The length of a branch represents the genetic distance between two nodes or between a node and a taxon, i.e., the number of mutations accumulated since divergence. The root corresponds to the common ancestor of all the taxa.</p>
</div>
<p><img alt="Figure 2" src="../img/phylo_1_2.jpg" /></p>
<p>Since the sequences represent viruses sampled from different individuals, a clade in the tree corresponds to a transmission chain. The number of sequences in a clade reflects the number of infections sampled from that transmission chain. The root of the tree corresponds to the origin of the epidemic.</p>
<p>After looking at your tree with <code>figtree</code>, can you answer these questions:</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Q1</label><label for="__tabbed_1_2">Answer</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>How many times was H1N1/2009 introduced in England during the outbreak?</p>
</div>
<div class="tabbed-block">
<p>Q1. The viruses sampled in England form 3 distinct clades, suggesting 3 introduction events (colored in red in the figure below). The first one involves 18 cases (top of the tree), the second 2 cases (A/England/247/2009 and A/England/251/2009), and the third one 3 cases (A/England/247/2009, A/England/312/2009, and A/England/415/2009).</p>
<p><img alt="Cluster 1" src="../img/phylo_1_Q1.png" /></p>
</div>
</div>
</div>
</div>
<div class="admonition question">
<p class="admonition-title">Question</p>
<div class="tabbed-set tabbed-alternate" data-tabs="2:2"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><div class="tabbed-labels"><label for="__tabbed_2_1">Q2</label><label for="__tabbed_2_2">Answer</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>What is/are the most likely geographical origins of the English strain(s)? (Take the sampling dates into account in your reasoning)</p>
</div>
<div class="tabbed-block">
<p>Q2. Cluster 1 is closest to a Canadian clade made of 4 cases. The two clades share a recent common ancestor located at node 1. All Canadian isolates were sampled in April 2009, while the viruses in the English Cluster 1 were sampled in May or June the same year. We can, therefore, assume that the strain migrated from the Americas to Europe (as we know it did…). We can also infer precedence of the Canadian outbreak by the fact that the Canadian sequences are genetically more similar to the common ancestor (node 1) than the English ones, as indicated by shorter branch lengths.</p>
<p>Cluster 2 is closest to a Canadian isolate (A/Toronto/3141/2009). Their common ancestor is located at node 2. The Toronto isolate predates the English ones, suggesting an introduction from Canada.</p>
<p>Cluster 3 is closest to a US isolate (A/New York/3532/2009). Their common ancestor is located at node 3. Again, we can conclude from the time of sampling and branch lengths that Clade 3 was imported from the US.</p>
</div>
</div>
</div>
</div>
<h2 id="dating-the-introductions-of-h1n109-in-england">Dating the Introduction(s) of H1N1/09 in England</h2>
<p>We will now estimate the time frame of these migration events. This time frame can be inferred from the rate at which mutations are accumulated in gene sequences. If, for instance, two sequences differ by 4 mutations and their rate of evolution is known to be around one substitution per nucleotide position per year, then 2 years have elapsed since they diverged from their common ancestor (i.e., 1 substitution x 2 years x 2 sequences = 4 mutations).</p>
<p>The first step of this procedure involves testing whether the genomes in the tree have evolved at a constant rate over time or not. This is called testing the <strong>molecular clock hypothesis</strong>.</p>
<p>If the molecular clock is constant or <strong>strict</strong>, the genetic distance between two sequences will be proportional to the time since these sequences last shared a common ancestor (as in the example given above). If the molecular clock is not constant, the correlation between genetic distance and time since divergence is weaker. The molecular clock is then said to be <strong>relaxed</strong>. Assuming a strict or relaxed molecular clock will have an impact on the dating of phylogenetic nodes. We will, therefore, test the molecular clock hypothesis before dating the tree.</p>
<h2 id="molecular-clock-testing">Molecular Clock Testing</h2>
<p>Open the software TempEst by running the following command:</p>
<div class="highlight"><pre><span></span><code>tempest
</code></pre></div>
<p>TempEst [2] is a tool designed to investigate the ‘temporal signal’ of molecular phylogenies. It tests whether there is sufficient genetic change between sampling times to reconstruct a statistical relationship between genetic divergence and time, which is the essence of a molecular clock in your data. You can download the software for free from <a href="http://tree.bio.ed.ac.uk/software/tempest">http://tree.bio.ed.ac.uk/software/tempest</a>.</p>
<p><u>How does it work?</u> TempEst performs ‘root-to-tip’ linear regressions, which can be used as a simple diagnostic tool for molecular clock models. It implies plotting the genetic divergence of the sequences (i.e., the sum of the branch lengths from a sequence - the tip - to the root of the tree) against the sampling time of the sequences (Figure 3). A linear trend with few residuals indicates that evolution follows a strict molecular clock. The same trend with greater scatter from the regression line suggests that a relaxed molecular clock model may be most appropriate. No trend at all indicates that the data contains little temporal signal and is unsuitable for inference using phylogenetic molecular clock models.</p>
<p><img alt="Figure 3" src="../img/phylo_1_3.jpg" /></p>
<p>Import the ML tree you saved earlier or, if you can’t find it anymore, select the file named <strong>RAxML_bipartitions.H1N1.flu.2009.ML.tre</strong> in your folder. If you are not prompted for the tree when <code>TempEst</code> opens, import it using the drop-down menu: <code>File</code> &gt; <code>Open…</code></p>
<p>We now have to tag the sequences with their sampling date. A list of all the sequences in the tree will appear (in the default <code>Sample Dates</code> tab). In order to plot root-to-tip genetic distances against sampling time, each sequence has to be associated with its date of sampling. Sampling dates are indicated at the end of the sequences’ name, preceded by the suffix ‘_d’, in the format YYYY-MM-DD. For instance, the sequence named ‘A/Lima/WRAIR1687P/2009_d2009-06-27’ was sampled on the 27th of June 2009.</p>
<p>In the Sample Date tab, click on <code>Parse Dates</code>. In the <code>Parse Dates for Taxa</code> window, select the following options:
- The date is given by a numerical field in the taxon label that is defined by a prefix and its order (Order: <strong>Last</strong>; Prefix: <strong>_d</strong>)
- Parse as a calendar date (Date format: <strong>yyy-MM-dd</strong>)</p>
<p>Click <code>OK</code>, and the time of sampling will appear in the <strong>Date</strong> column.</p>
<p>Go to the <code>Root-to-tip</code> tab to see the root-to-tip linear regression plot. Check the box marked <strong>Best-fitting root</strong>. Summary statistics of the plot are shown in the left-hand side window:</p>
<ul>
<li><strong>Date range</strong>: the maximum time interval between two sampled sequences.</li>
<li><strong>Slope</strong>: The slope of the regression line; corresponds to the rate of evolution, here the average number of nucleotide substitutions per unit of time (here, per year).</li>
<li><strong>X-intercept</strong>: the time at which the viral population had no genetic diversity, which corresponds to the time of the most recent common ancestor of the sampled population (tMRCA).</li>
<li><strong>Correlation Coefficient</strong>: A measure of the relationship between time and the number of accumulated mutations (range: -1, 1). A positive/negative value implies a positive/negative linear relationship between time and diversity (i.e., as time increases, so does genetic divergence). A value close to 0 suggests no relationship between time and genetic divergence.</li>
<li><strong>R squared</strong>: A measure of how close the data is to the regression line, i.e., what proportion of the variation in genetic divergence is explained by a strict molecular clock hypothesis.</li>
</ul>
<div class="admonition question">
<p class="admonition-title">Question</p>
<div class="tabbed-set tabbed-alternate" data-tabs="3:2"><input checked="checked" id="__tabbed_3_1" name="__tabbed_3" type="radio" /><input id="__tabbed_3_2" name="__tabbed_3" type="radio" /><div class="tabbed-labels"><label for="__tabbed_3_1">Q3</label><label for="__tabbed_3_2">Answer</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>Can we assume that the rate of evolution of the viruses in the phylogeny is constant over time (i.e., that the molecular clock is strict)?</p>
</div>
<div class="tabbed-block">
<p>The root-to-tip linear regression plot shows a positive correlation between time and divergence (correlation coefficient = 0.71; see the figure above). This suggests that the phylogeny has sufficient temporal signal to conduct a molecular clock analysis on this dataset.</p>
<p>However, there is a certain degree of scatter from the regression line, and only 51% of the variation in genetic divergence is explained by a strict molecular clock hypothesis (R squared = 0.51). We will, therefore, assume significant variation from a strict molecular clock in this dataset and opt for a relaxed molecular clock (i.e., the rate of evolution varies over time and across lineages). This is despite a relatively short sampling interval (about 3 months) and moderate genetic diversity in the sampled population.</p>
<p><img alt="Figure 3" src="../img/phylo_1_Q3.png" /></p>
</div>
</div>
</div>
</div>
<h2 id="dating-migration-events">Dating Migration Events</h2>
<p>Now that the properties of the molecular clock in our dataset have been established, we will estimate the time(s) at which H1N1/09 was introduced in England.</p>
<div class="admonition reminder">
<p class="admonition-title">Reminder</p>
<p>In a ‘classic’ phylogenetic tree, branch lengths reflect as the number of nucleotide substitutions per site. Rates of evolution are expressed as the number of substitutions per site per unit of time. By dividing the length of a branch by the rate of evolution, we end up with a tree where branch lengths represent time units. A branch therefore represents the time elapsed between two nodes. Or, in our case, the time elapsed between two transmission events.</p>
</div>
<p>The reconstruction of dated phylogenetic trees is computationally intense and could not be achieved within the time imparted for this practical. A dated H1N1/09 phylogeny was therefore built prior to the session, under the appropriate molecular clock model, using the Bayesian MCMC approach implemented in the software <strong>BEAST</strong> v.1.8 [3]. The resulting dated tree is in your folder under the name <strong>H1N1.flu.2009.mol_clock.tre</strong>. A more detailed example of how to set up a BEAST v1.8. analysis, including producing the input files and analysing the output is shown in the <strong>Appendix</strong>.</p>
<p>How does it work? Each tip of the tree has a known time, given by the sampling date of the sequence. Internal nodes are given arbitrary starting times consistent with their order in the tree (from the tips to the root). An additional parameter, the evolution rate, is used to scale these times into expected number of nucleotide substitutions per site. Markov chain Monte Carlo integration is then used to summarize the probability density function of a model tested against the data, providing a representative sample of parameter values of the chosen model. The model includes the tree topology, the times of internal nodes and the evolution rate.</p>
<p>We will use the program <strong>FigTree</strong> to display the dated tree and its annotations.</p>
<p>Open <strong>FigTree</strong> by running the command:
<div class="highlight"><pre><span></span><code>figtree
</code></pre></div></p>
<p>Import the dated tree: <code>File</code>&gt; <code>Open…</code> &gt; <strong>H1N1.flu.2009.mol_clock.2.tre</strong></p>
<p>A phylogeny will appear. Again, go to <code>Tree</code> &gt; <code>Increasing Node Order</code> to display the tree in the same way as the one you generated with RAxML. This will ease comparison. Note that the major clusters do not change, just the order in which the clusters are organised from top to bottom. The dated tree should be very similar to the ML tree you reconstructed in Session 1. However, in this tree, the branch lengths represent days rather than genetic distances. Notice the scale at the bottom of the tree.</p>
<p>On the lefthand side toolbar, tick the <code>Node Labels</code> box. The age of the nodes will appear.</p>
<p>The age of a node is expressed as the number of days prior to the most recent sampling date in the tree. Here, the most recent sample is A/Lima/WRAIR1687P/2009, sampled on the 27th of June 2009. If, for instance, a node age equals 21 days, it means that the date at this node is the 6th of June 2009.</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
<div class="tabbed-set tabbed-alternate" data-tabs="4:2"><input checked="checked" id="__tabbed_4_1" name="__tabbed_4" type="radio" /><input id="__tabbed_4_2" name="__tabbed_4" type="radio" /><div class="tabbed-labels"><label for="__tabbed_4_1">Q4</label><label for="__tabbed_4_2">Answer</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>According to the molecular clock dating, what are the date(s) of introduction of H1N1/09 in England (rounded down)?
Tip: If mental arithmetic is not your thing, you can use the 2009 calendar provided as Appendix to back-calculate the date of the nodes.</p>
</div>
<div class="tabbed-block">
<p>The most recent viral sequence (A/Lima/WRAIR1687P/2009) was sampled on 27/06/2009. The ancestral node of Cluster 1 has an age of 63 days (rounded down) prior to the most recent sample. The estimated date of the first introduction of H1N1/09 in England is thus 25/04/2009.</p>
<p>The ancestral node of Cluster 2 has an age of 57 days (rounded down) prior to the most recent sample. The estimated date of the second introduction of H1N1/09 in England is thus 01/05/2009.</p>
<p>The ancestral node of Cluster 3 has an age of 40 days (rounded down) prior to the most recent sample. The estimated date of the third introduction of H1N1/09 in England is thus 18/05/2009.</p>
</div>
</div>
</div>
</div>
<h2 id="identifying-the-geographical-origins-of-the-h1n109-strains-imported-in-england">Identifying the geographical origin(s) of the H1N1/09 strains imported in England</h2>
<p>We will finally reconstruct the migration pathways of these H1N1/09 strains.</p>
<div class="admonition reminder">
<p class="admonition-title">Reminder</p>
<p>When individuals are infected in one location and then move to another, or infect someone whilst travelling, this is apparent as a “change” in the location ascribed to one branch of the tree. These changes in location along a phylogenetic tree can be inferred from the location values at the tips and the shape of the tree (see Figure 4). To do so, a model of location exchange process is fitted to the data and the most likely location of the viral strain positioned at the nodes of the tree, together with its probability, can be estimated using a MCMC sampling procedure similar to the one used in Section 2.</p>
</div>
<p><img alt="Figure 3" src="../img/phylo_1_4.jpg" /></p>
<p>This approach was applied to the H1N1/09 phylogeny, using an asymmetric continuous-time Markov chain [5], as implemented in the program BEAST. The asymmetric model uses separate parameters for forward and reverse rates of movement between each pair of locations. The tree file we used for the molecular clock analysis (<strong>H1N1.flu.2009.mol_clock.2.tre</strong>) also contains the result of the inferred migration patterns.</p>
<p>On the left hand side toolbar, pull down the <code>Node Labels</code> menu and under<code>Display</code>, select <strong>Location</strong>. The most likely location of the ancestral virus located at the nodes will appear.</p>
<p>To<code>display</code> the posterior probability of the most likely node location, select <strong>Location.prob</strong> in the Display menu of the <code>Node Labels</code>.</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
<div class="tabbed-set tabbed-alternate" data-tabs="5:2"><input checked="checked" id="__tabbed_5_1" name="__tabbed_5" type="radio" /><input id="__tabbed_5_2" name="__tabbed_5" type="radio" /><div class="tabbed-labels"><label for="__tabbed_5_1">Q5</label><label for="__tabbed_5_2">Answer</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>Which country is the most likely source of the H1N1/09 epidemic? What is the probability of that location?</p>
</div>
<div class="tabbed-block">
<p>The most likely location state at the root of the tree is the US (posterior probability: 0.84).</p>
</div>
</div>
</div>
</div>
<div class="admonition question">
<p class="admonition-title">Question</p>
<div class="tabbed-set tabbed-alternate" data-tabs="6:2"><input checked="checked" id="__tabbed_6_1" name="__tabbed_6" type="radio" /><input id="__tabbed_6_2" name="__tabbed_6" type="radio" /><div class="tabbed-labels"><label for="__tabbed_6_1">Q6</label><label for="__tabbed_6_2">Answer</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>Where were the English strains of H1N1/09 imported from? How confident are we? How does your answer compare to that of Q4?**</p>
</div>
<div class="tabbed-block">
<ul>
<li>The location at the node from which Cluster 1 is branching off (node 1 in the figure shown with the answer to Q1) is the US (posterior probability: 0.55).</li>
<li>The location at the node from which Cluster 2 is branching off (node 2 in the figure shown with the answer to Q1) is Canada (posterior probability: 0.48).</li>
<li>The location state at the node from which Cluster 3 is branching off (node 3 in the figure shown with the answer to Q1) is the US (posterior probability: 0.87).</li>
</ul>
<p>The geographical origin of the founder viruses is consistent for Cluster 2 and 3. However, for Cluster 1, the tree topology suggests relatedness with a Canadian cluster while the phylogeographic reconstruction places the founder virus of that cluster in the US. The most likely explanation is that the English and Canadian viruses were both imported from the US.</p>
</div>
</div>
</div>
</div>
<h3 id="references">References</h3>
<ol>
<li>A. Stamatakis: 'RAxML Version 8: A tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies'. <em>Bioinformatics</em>, 2014</li>
<li>A. Rambaut, T. T. Lam, L. Max Carvalho, and O. G. Pybus, ‘Exploring the temporal structure of heterochronous sequences using TempEst (formerly Path-O-Gen)’, <em>Virus Evol.</em>, vol. 2, no. 1, p. vew007, Jan. 2016.</li>
<li>A. J. Drummond, M. A. Suchard, D. Xie, and A. Rambaut, ‘Bayesian Phylogenetics with BEAUti and the BEAST 1.7’, <em>Mol. Biol. Evol.</em>, vol. 29, no. 8, pp. 1969–1973, Jan. 2012.</li>
<li>N. R. Faria et al., ‘The early spread and epidemic ignition of HIV-1 in human populations’, <em>Science</em>, vol. 346, no. 6205, pp. 56–61, Mar. 2014.</li>
<li>P. Lemey, A. Rambaut, A. J. Drummond, and M. A. Suchard, ‘Bayesian phylogeography finds its roots’, <em>PLoS Comput. Biol.</em>, vol. 5, no. 9, p. e1000520, Sep. 2009.</li>
</ol>
<h2 id="appendix">Appendix</h2>
<h3 id="phylogenetic-tree-building-with-beast">Phylogenetic Tree Building with BEAST</h3>
<p>(Note: This example is for initializing a BEAST v.1.8 analysis; BEAST2 has a different interface.)</p>
<p><strong>1. Creating a BEAST Input File Using BEAUti</strong></p>
<p>Launch BEAUti by typing <code>beauti</code> into the command line or opening it from the dashboard.</p>
<p>1.1. Load the FASTA sequence file <strong>H1N1.flu.2009.fas</strong> by clicking the <code>+</code> button or going to <code>File</code> &gt; <code>Open…</code>.</p>
<p><img alt="BEAUti Sequence" src="../img/phylo_1_5.jpg" /></p>
<p>This first will give you information on the number of taxa (50), the number of sites (12735), and the data type (nucleotide).</p>
<p>If you want to partition the data, i.e. if you have samples that you know are from different populations or lineages, you can go onto the ‘Taxa’ tab and set these partitions. This will allow you to specify whether these groups are monophyletic as well as divergence dates between groups. For this exercise, we will not assume any partitioning of samples.</p>
<p>1.2. Attach a date to each sample.</p>
<p>This can be done in the same way as was shown in TempEst:
In the Tips tab, tick the box for <strong>Use tip dates</strong>, and then click on <strong>Guess Dates</strong>. In the <strong>Guess date values for all taxa</strong> window, select the following options:</p>
<ul>
<li>The date is given by a numerical field in the taxon label that is defined by a prefix and its order (Order: Last; Prefix: _d)</li>
<li>Parse as a calendar date (Date format: <strong>yyyy-MM-dd</strong>)
Alternatively, you can create a tab-delimited file (.txt) with two columns corresponding to the taxa names and sample dates, and use <strong>Import Date</strong>s.</li>
</ul>
<p><img alt="BEAUti Substitution Model" src="../img/phylo_1_6.jpg" /></p>
<p>1.3. Set the location for each sample</p>
<p>To attach further traits to each sample, such as location, in the <code>Traits</code> tab, click on <code>Import Traits</code> and open a tab-delimited file (.txt) with two columns corresponding to the taxa names and location (or characteristic of choice).</p>
<p>1.4. Set a site (substitution) model. </p>
<p>This allows us to substitution and site heterogeneity models. The rate and likelihood of base substitution over the genome will be variable so we will set some parameters to describe the model of substitution. We will set the substitution model to GTR – Generalized Time Reversible model – as this is the most frequently used model and allows for variable base frequencies. It is also reversible (i.e. T-&gt;A same rate as A-&gt;T). We can also set the site heterogeneity model, i.e. the frequency of each site to a <strong>Gamma</strong> distribution with 4 categories.</p>
<p>In the Sites tab, select <strong>GTR</strong> from the ‘<strong>Substitution model</strong>’ dropdown box. We will set the ‘<strong>Site Heterogeneity model</strong>’ as ‘<strong>Gamma</strong>’ with 4 categories.</p>
<p><img alt="BEAUti Substitution Model" src="../img/phylo_1_7.jpg" /></p>
<p>1.5 Set a clock model.
The <strong>‘Clocks’</strong> tab will allow us to set the molecular clock for mutations along each branch of the tree. If we set this to <strong>‘Strick clock’</strong> this specifies that the mutation rate along each branch will be invariable. As we have shown in the TempEst analysis that our data varies significantly from a strict molecular clock, we will select a relaxed clock to allow for rate variation across branches in our tree.</p>
<p>In the Clocks tab, select <strong>‘Uncorrelated relaxed clock’</strong> from the <strong>‘Clock Type’</strong> dropdown box and keep the <strong>‘Relaxed Distribution’</strong> option as <strong>‘Lognormal</strong>’.</p>
<p><img alt="BEAUti Substitution Model" src="../img/phylo_1_8.jpg" /></p>
<p>1.6 Set a tree model.</p>
<p>As the phylogenetic tree is built through a coalescent approach, backwards through time based on the genetic diversity at tips/samples, you can will need to set a model for the demographic history of the population that describes the effective population size through time. For example, <strong>‘Constant size’</strong> will state that the effective population size has remained constant through evolutionary time. For the purposes of this exercise, we will choose a constant population size.</p>
<p>In the <code>Trees</code> tab, select <strong>‘Coalescent: Constant Size’</strong> from the <strong>‘Tree Prior’</strong> dropdown box.</p>
<p><i> Note: You can also input a starting tree on which BEAST will base the topology of the Bayesian tree. This may be a neighbor-joining or parsimony tree and can speed up the analysis by approximating the topology as a prior. We will start with a random tree.</i></p>
<p>The '<strong>States</strong>', '<strong>Priors</strong>’ and '<strong>Operators</strong>' tabs allow us to specify whether to reconstruct sequences, set specific priors for each parameter, and to specify which operators to include in the output log file. We will leave all these as default values.</p>
<p>1.7 Define the length of the MCMC chain</p>
<p>Finally, the <strong>‘MCMC’</strong> tab is where we will define the steps in the MCMC chain that is used to construct trees. Here we are interested in the length of the chain. This is the number of iterations (number of steps in the random walk) the program will go through to converge on the best tree topology. The higher this number, the more support there will be for the final tree and the more chance of convergence.</p>
<p>Define the length of the chain by typing '<strong>10000000</strong>' in the '<strong>Length of chain</strong>' box.</p>
<p>The name of the output files can be specified here. This will determine the names of the files that will be created after the BEAST run, .log, .trees, and .ops. The output file from BEAUti will be a <strong>.xml</strong> file for BEAST input. We will call this H1N1.flu.2009, so type this in the ‘<strong>File name stem</strong>’ box, if it doesn’t already appear there.</p>
<p><img alt="BEAUti Substitution Model" src="../img/phylo_1_9.jpg" /></p>
<p>Then click on the <code>Generate BEAST file</code> button at the bottom right of the screen to generate the .xml file that will be used as the input for BEAST.</p>
<ol>
<li>Running the BEAST analysis
BEAST can be run on the command line or through the desktop application. We will run BEAST on command line.</li>
</ol>
<p>To run BEAST, open up a terminal and make sure you are in the folder containing your .xml file from the BEAUti output and type:
<div class="highlight"><pre><span></span><code>beast H1N1.flu.2009.xml
</code></pre></div>
This will create three files; .log, .trees, and .ops, with the “H1N1.flu.2009.” prefix.</p>
<ol>
<li>Evaluating your BEAST run
Open the program TRACER by typing <code>tracer</code>. This will allow you to check the output from BEAST and determine how effectively the MCMC has mixed, i.e. how close it is to reaching a steady state (converging). This can be used to manually evaluate the performance of the BEAST run.</li>
</ol>
<p>Load in the ‘<strong>H1N1.flu.2009.log</strong>’ file to Tracer to see the output.</p>
<p><img alt="BEAUti Substitution Model" src="../img/phylo_1_10.jpg" /></p>
<p><img alt="BEAUti Substitution Model" src="../img/phylo_1_11.jpg" /></p>
<p>Here we want to look at the <strong>ESS score</strong>. This is the Effective Sample Size, which is a measure of the number of effectively independent draws from the posterior distribution that the Markov chain is equivalent to. Low scores are highlighted in red, with scores &gt;200 considered sufficient to demonstrate good mixing.</p>
<p>To increase ESS scores, you can run the analysis for longer (more MCMC iterations), or do multiple independent runs that can be combined with the program <strong>LogCombiner</strong>. Very low scores for many parameters may though suggest that the prior parameters are not a good fit and should be varied.</p>
<p>In our data, we can see that most values are over 200, with only the prior and posterior values highlighted in red. Looking at the “Trace” of the MCMC though, we can see that it remaining reasonably steady around -19600 for the prior parameter, suggesting that this support will increase (ESS will increase) if the MCMC chain was longer.</p>
<ol>
<li>Producing a single phylogenetic tree</li>
</ol>
<p>The program <strong>TreeAnnotator</strong> summarises the sample of trees produced by BEAST (in the .tree file) onto a single maximum clade credibility tree. The posterior probabilities of the nodes in the target tree, the posterior estimates and HPD limits of the node heights and (in the case of a relaxed molecular clock model) the rates, are shown.</p>
<p>Load in the ‘<strong>H1N1.flu.2009.trees</strong>’ file to TreeAnnotator through <code>Choose File…</code> in <code>Input Tree File</code>. Select what to the call the output file, e.g. <strong>H1N1.flu.2009.tre</strong></p>
<p>You can also set the number of burnin states (The number of initial MCMC runs to discard).</p>
<p>Then select <code>Run</code>to produce a single maximum clade credibility tree.
<img alt="BEAUti Substitution Model" src="../img/phylo_1_12.jpg" /></p>
<hr>
<h2 id="calendar">Calendar</h2>
<p><img alt="BEAUti Substitution Model" src="../img/phylo_1_13.jpg" /></p>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["content.tooltips", "navigation.instant", "navigation.tabs", "toc.integrate", "content.tabs.link"], "search": "../assets/javascripts/workers/search.f8cc74c7.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.c8b220af.min.js"></script>
      
    
  </body>
</html>